pages/partials/question.ejs

```diff
@@ -15,10 +15,37 @@
 
     <table class="table table-sm table-hover two-column-description">
       <tbody>
-        <% if (devMode || authz_data.has_instructor_view) {
-              var body = encodeURIComponent("Hello " + issue.user_name + ",\n\nRegarding the issue of:\n\n\"" + (issue.student_message || '-') +"\"\n\nWe've..."); %>
+        <% if (
+                devMode ||
+                (!issue.course_instance_id) ||
+                (
+                  locals.course_instance &&
+                  (course_instance.id == issue.course_instance_id) &&
+                  authz_data.has_course_instance_permission_view
+                )
+              ) {
+
+          // There are three situations in which the issue need not be anonymized:
+          //
+          //  1) We are in devMode.
+          //
+          //  2) The issue is not associated with a course instance. The only way
+          //     for a user to generate an issue that is not associated with a course
+          //     instance is if they are an instructor, so there are no student data
+          //     to be protected in this case.
+          //
+          //  3) We are accessing this page through a course instance, the issue is
+          //     associated with the same course instance, and the user has student
+          //     data view access.
+          //
+          // Otherwise, all issues must be anonymized.
+
         %>
-        <tr><th>Student:</th><td><%= issue.user_name || '-'  %> (<a href="mailto:<%= issue.user_uid  || '-' %>?subject=Reported%20PrairieLearn%20Issue&body=<%= body %>"><%= issue.user_uid || '-' %></a>)</td></tr>
+        <% let body = encodeURIComponent("Hello " + issue.user_name + ",\n\nRegarding the issue of:\n\n\"" + (issue.student_message || '-') +"\"\n\nWe've..."); %>
+        <tr><th>User:</th><td><%= issue.user_name || '-'  %> (<a href="mailto:<%= issue.user_uid  || '-' %>?subject=Reported%20PrairieLearn%20Issue&body=<%= body %>"><%= issue.user_uid || '-' %></a>)</td></tr>
```

Comment:
To resolve a conflict as part of the merge 0e0228508d961a74056cb24a13cefeb57788cec4, I added the `let body = ` line and used it as `<%= body %>` in the line above. This change was introduced in PR #3314 (commit 61b7214fe3b22eb4f24ae2164ab0e8b7feee8a01).